Rest/Content/ContentType.php

```diff
@@ -0,0 +1,379 @@
+        )->searchHits;
+    }
+
+    /**
+     * Returns paged content based on ContentType ids.
+     *
+     * @param int $contentTypeId
+     * @param array $options
+     *
+     * @return int
+     */
+    private function countContentByContentTypeId($contentTypeId, $options)
+    {
+        $criteria = array(new Criterion\ContentTypeId($contentTypeId));
+
+        if ($options['path']) {
+            $criteria[] = new Criterion\Subtree($options['path']);
+        }
+
+        if (!$options['hidden']) {
+            $criteria[] = new Criterion\Visibility(Criterion\Visibility::VISIBLE);
+        }
+
+        $criteria[] = $this->generateSubtreeCriteria($options);
+
+        $query = new Query();
+        $query->query = new Criterion\LogicalAnd($criteria);
+        $query->limit = 0;
+
+        return $this->searchService->findContent($query)->totalCount;
+    }
+
+    private function generateSubtreeCriteria($options)
+    {
+        if (!empty($options['mandatorId'])) {
+            $siteaccesses = $this->siteAccessHelper->getSiteaccessesByMandatorId($options['mandatorId']);
+        } elseif (!empty($options['siteAccess'])) {
+            $siteaccesses = array($options['siteAccess']);
+        } else {
+            $siteaccesses = array($this->siteAccess->name);
+        }
+
+        $subtreeCriteria = [];
+        $rootLocations = $this->siteAccessHelper->getRootLocationsBySiteaccesses($siteaccesses);
+        foreach ($rootLocations as $rootLocationId) {
+            $subtreeCriteria[] = new Criterion\Subtree($this->locationService->loadLocation($rootLocationId)->pathString);
+        }
+
+        return new Criterion\LogicalOr($subtreeCriteria);
+    }
+
+    /**
+     * Returns languages based on mandatorId or siteaccess.
+     *
+     * @param array $options
+     *
+     * @return array
+     *
+     * @throws LogicException When languages cannot be fetched from siteAccess or mandatorId.
+     */
+    private function getLanguages(array $options)
+    {
+        if (!empty($options['mandatorId'])) {
+            $languages = $this->siteAccessHelper->getMainLanguagesBySiteaccesses(
+                $this->siteAccessHelper->getSiteaccessesByMandatorId($options['mandatorId'])
+            );
+        } elseif (!empty($options['siteAccess'])) {
+            $languages = $this->configResolver->getParameter('languages', '', $options['siteAccess']);
+        } else {
+            $languages = $this->configResolver->getParameter('languages');
+        }
+
+        if (empty($languages)) {
+            throw new LogicException(sprintf('No languages found using siteAccess or mandatorId'));
+        }
+
+        return $languages;
+    }
+
+    /**
+     * Generate export files.
+     *
+     * @param $languages
+     * @param $chunkDir
+     * @param array $options
+     *
+     * @return array
+     */
+    private function generateFiles($languages, $chunkDir, array $options)
```

Comment:
It doesn't really "generate a file". Maybe it "prepares" something ? or "build" some kind of list ?

Reply:
Actually, after digging further, it does. So nevermind this.